CalendarApp/Notification Handler/NotificationHandler.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  NotificationHandler.swift
+//  CalendarApp
+//
+//  Created by Angelina Zhang on 8/4/22.
+//
+
+import Foundation
+import UserNotifications
+import CoreData
+
+@objcMembers
+class NotificationHandler : NSObject {
+    private let center = UNUserNotificationCenter.current()
+    private let context = (UIApplication.shared.delegate as! AppDelegate).persistentContainer.viewContext
+    private var calendar = Calendar(identifier: .gregorian)
+    
+    override init() {
+        super.init()
+        
+        calendar.timeZone = TimeZone.current
+    }
+    
+    func registerNotifications(with completion: @escaping() -> Void) {
+        center.requestAuthorization(options: [.alert, .badge, .sound]) { (granted, error) in
+            completion()
+        }
+    }
+    
+    func deleteReminderWithID(_ reminderID: UUID) {
+        let request = Reminder.fetchRequest()
+        request.predicate = NSPredicate(format: "reminderID == %@",
+                                        reminderID as CVarArg)
+        do {
+            let notifications = try context.fetch(request)
+            if notifications.count == 1 {
+                context.delete(notifications[0])
+                guard let reminderID = notifications[0].reminderID else {
+                    return
+                }
+                center.removePendingNotificationRequests(withIdentifiers: [reminderID.uuidString])
+                try context.save()
+            }
+        } catch {
+            // TODO: error handling
+        }
+    }
+    
+    func fetchReminderInfo() -> [Reminder] {
+        let request = Reminder.fetchRequest()
+        request.predicate = NSPredicate(format: "archived == false")
+        do {
+            let reminders = try context.fetch(request)
+            return reminders
+        } catch {
+            return [Reminder]()
+        }
+    }
+    
+    func updateReminderForEvent(_ event: Event, _ date: Date) {
+        let request = Reminder.fetchRequest()
+        request.predicate = NSPredicate(format: "eventID == %@",
+                                        event.objectUUID as CVarArg)
+        do {
+            let notifications = try context.fetch(request)
+            if notifications.count == 1 {
+                context.delete(notifications[0])
+                guard let reminderID = notifications[0].reminderID else {
+                    return
+                }
+                center.removePendingNotificationRequests(withIdentifiers: [reminderID.uuidString])
```

Comment:
Using the reminder UUID this way is false reuse. This object should have a separate field for notificationIDs imo. If reminderID needs to change it shouldn't affect notifications.